lib/backend/model/bgppeer.go

```diff
@@ -0,0 +1,149 @@
+// Copyright (c) 2016 Tigera, Inc. All rights reserved.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package model
+
+import (
+	"fmt"
+	"regexp"
+
+	"reflect"
+
+	"github.com/golang/glog"
+	"github.com/tigera/libcalico-go/lib/errors"
+	"github.com/tigera/libcalico-go/lib/net"
+	"github.com/tigera/libcalico-go/lib/scope"
+)
+
+var (
+	matchGlobalBGPPeer = regexp.MustCompile("^/?calico/bgp/v1/global/peer_v./([^/]+)$")
+	matchHostBGPPeer   = regexp.MustCompile("^/?calico/bgp/v1/host/([^/]+)/peer_v./([^/]+)$")
+	typeBGPPeer        = reflect.TypeOf(BGPPeer{})
+)
+
+type BGPPeerKey struct {
+	Scope    scope.GlobalOrNode `json:"-" validate:"omitempty"`
+	Hostname string             `json:"-" validate:"omitempty"`
+	PeerIP   net.IP             `json:"-" validate:"required"`
+}
+
+func (key BGPPeerKey) DefaultPath() (string, error) {
+	if key.PeerIP.IP == nil {
+		return "", errors.ErrorInsufficientIdentifiers{Name: "peerIP"}
+	}
+	switch key.Scope {
+	case scope.Global:
+		e := fmt.Sprintf("/calico/bgp/v1/global/peer_v%d/%s",
+			key.PeerIP.Version(), key.PeerIP)
+		return e, nil
+	case scope.Node:
+		if key.Hostname == "" {
+			return "", errors.ErrorInsufficientIdentifiers{Name: "hostname"}
+		}
+		e := fmt.Sprintf("/calico/bgp/v1/host/%s/peer_v%d/%s",
+			key.Hostname, key.PeerIP.Version(), key.PeerIP)
+		return e, nil
+	}
+	panic(fmt.Errorf("Unexpected scope value: %d", key.Scope))
```

Comment:
I think that is right.  It might not be possible through calicoctl, but that doesn't mean that it won't be possible through all possible future users of this interface.